relay/README.md

```diff
@@ -24,18 +24,24 @@ libp2p already has modules for NAT ([go-libp2p-nat](https://github.com/libp2p/go
 but these don't always do the job, just because NAT traversal is complicated.
 That's why it's useful to have a simple relay protocol.
 
-One node asks a relay node to connect to another node on its behalf.
-The relay node shortcircuits its streams to the two nodes,
+Unlike a transparent **tunnel**, where a libp2p peer would just proxy a communication
+stream to a destination (the destination being unaware of the original source),
+a circuit-relay makes the destination aware of the original source and
+the circuit followed to stablish communication between the two. This provides
+the destination side with the capacity of re-contacting the original source
+by rebuilding the circuit in the oposite direction, using the same relay.
```

Comment:
@hsanjuan @lgierth can this part be clarified a little further, because the wording is explicit in saying that the connection is rebuilt using the same relay, which makes me ask why can't it be rebuilt using an alternative relay, in case the original one is unavailable. _I guess this is just wording, and in case the original is unavailable a new one will be selected, or not?_